lua/esqueleto/utils.lua

```diff
@@ -35,23 +35,39 @@ end
 ---@param file string Template file path
 ---@param opts table Plugin configuration table
 M.writetemplate = function(file, opts)
-  if file ~= nil and not opts.wildcards.expand then
-    -- Place contents of template directly to buffer
-    vim.cmd("0r " .. file)
-    vim.cmd("norm G")
-  elseif file ~= nil and opts.wildcards.expand then
-    -- Expand wildcards from template and place contents in buffer
-    local content = io.open(file, "r"):read("*a")
-    local parsed_content, cursor_pos = wildcards.parse(content, opts.wildcards.lookup)
-    if content ~= nil then
-      vim.api.nvim_buf_set_lines(0, 0, -1, true, parsed_content)
-    end
-    -- If a cursor wildcard was found, place cursor there
-    if cursor_pos ~= nil then
-      vim.api.nvim_win_set_cursor(0, cursor_pos)
-    else
-      vim.cmd("norm G")
-    end
+  if file == nil then
+    -- Do an early return if no files are specified
```

Comment:
I've added comments to each logic. If you find these a bit noisy, let me know and I will reduce them to the bare minimum.